pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -155,47 +185,49 @@ func (r *Reconciler) reconcileMachineWithCloudState(failedCondition *v1beta1.GCP
 	if failedCondition != nil {
 		r.providerStatus.Conditions = reconcileProviderConditions(r.providerStatus.Conditions, *failedCondition)
 		return nil
-	} else {
-		freshInstance, err := r.computeService.InstancesGet(r.projectID, r.providerSpec.Zone, r.machine.Name)
-		if err != nil {
-			return fmt.Errorf("failed to get instance via compute service: %v", err)
-		}
+	}
+	freshInstance, err := r.computeService.InstancesGet(r.projectID, r.providerSpec.Zone, r.machine.Name)
+	if err != nil {
+		return fmt.Errorf("failed to get instance via compute service: %v", err)
+	}
+	return r.populateMachineWithInstance(freshInstance)
+}
 
-		if len(freshInstance.NetworkInterfaces) < 1 {
-			return fmt.Errorf("could not find network interfaces for instance %q", freshInstance.Name)
-		}
-		networkInterface := freshInstance.NetworkInterfaces[0]
+func (r *Reconciler) populateMachineWithInstance(instance *compute.Instance) error {
+	if len(instance.NetworkInterfaces) < 1 {
+		return fmt.Errorf("could not find network interfaces for instance %q", instance.Name)
+	}
+	networkInterface := instance.NetworkInterfaces[0]
 
-		nodeAddresses := []v1.NodeAddress{{Type: v1.NodeInternalIP, Address: networkInterface.NetworkIP}}
-		for _, config := range networkInterface.AccessConfigs {
-			nodeAddresses = append(nodeAddresses, v1.NodeAddress{Type: v1.NodeExternalIP, Address: config.NatIP})
-		}
-		// Since we don't know when the project was created, we must account for
-		// both types of internal-dns:
-		// https://cloud.google.com/compute/docs/internal-dns#instance-fully-qualified-domain-names
-		// [INSTANCE_NAME].[ZONE].c.[PROJECT_ID].internal (newer)
-		nodeAddresses = append(nodeAddresses, corev1.NodeAddress{
-			Type:    corev1.NodeInternalDNS,
-			Address: fmt.Sprintf("%s.%s.c.%s.internal", r.machine.Name, r.providerSpec.Zone, r.projectID),
-		})
-		// [INSTANCE_NAME].c.[PROJECT_ID].internal
-		nodeAddresses = append(nodeAddresses, corev1.NodeAddress{
-			Type:    "AltInternalDNS",
-			Address: fmt.Sprintf("%s.c.%s.internal", r.machine.Name, r.projectID),
-		})
+	nodeAddresses := []v1.NodeAddress{{Type: v1.NodeInternalIP, Address: networkInterface.NetworkIP}}
+	for _, config := range networkInterface.AccessConfigs {
+		nodeAddresses = append(nodeAddresses, v1.NodeAddress{Type: v1.NodeExternalIP, Address: config.NatIP})
+	}
+	// Since we don't know when the project was created, we must account for
+	// both types of internal-dns:
+	// https://cloud.google.com/compute/docs/internal-dns#instance-fully-qualified-domain-names
+	// [INSTANCE_NAME].[ZONE].c.[PROJECT_ID].internal (newer)
```

Comment:
It looks like you can force the new style on a per-project, or per-instance basis:
https://cloud.google.com/compute/docs/internal-dns#setting-zonal-dns

Should we set `VmDnsSetting=ZonalOnly` and avoid this?

Reply:
IIRC, didn't @michaelgugino mention that it depends on when your account was created. (I could have that recollection totally wrong.)

Reply:
The default depends on that, but I think you can override it.

Reply:
Sure, but we're not in charge of creating a user's project, that is same as an aws account.  Enforcing a project-wide setting is probably not an option for us.